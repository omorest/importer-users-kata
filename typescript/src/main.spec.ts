import {expect, describe, it} from "vitest";
import {ImportUsers} from "./ImportUsers.js";

class ImportUsersTestable extends ImportUsers {
  public readonly lines: string[] = []

  log(message: string) {
    this.lines.push(message)
  }
}

describe('Import users', function () {
  it('should import users from CSV and URL', async function () {
    const importUsers = new ImportUsersTestable()
    await importUsers.execute()
    expect(importUsers.lines).toMatchSnapshot()
  })
});
