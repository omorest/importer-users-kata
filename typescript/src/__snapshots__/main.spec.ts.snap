// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Import users > should import users from CSV and URL 1`] = `
[
  "*********************************************************************************",
  "* ID		* COUNTRY	* NAME		* EMAIL				*",
  "*********************************************************************************",
  "* 200189617246	* Germany	* Lukas Schmidt	* lukas.shmidt@example.com	*",
  "* 200189016257	* Germany	* Maria Fischer	* maria.fischer@example.com	*",
  "* 230573109005	* Spain 	* Luis Sanchez	* luis.sanchez@example.com	*",
  "* 230854119034	* Italy 	* Elio Pausini	* elio.pausini@example.com	*",
  "* 270054311737	* India 	* Mitesh Kumari	* mitesh.kumari@example.com	*",
  "* 202160712259	* Germany	* Elena Mueller	* elena.muller@example.com	*",
  "* 270554319031	* India 	* Natasha Shah	* natasha.shah@example.com	*",
  "* 100000000001	* Australia	* Nevaeh Dunn	* nevaeh.dunn@example.com	*",
  "* 100000000002	* Norway	* Sara Abdallah	* sara.abdallah@example.com	*",
  "* 100000000003	* France	* Melvin Perrin	* melvin.perrin@example.com	*",
  "* 100000000004	* Australia	* Dawn Snyder	* dawn.snyder@example.com	*",
  "* 100000000005	* Netherlands	* Irina Kaptein	* irina.kaptein@example.com	*",
  "*********************************************************************************",
  "12 users in total!",
]
`;
